csharp/src/SeedLang/Interpreter/VM.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using SeedLang.Common;
+using SeedLang.Runtime;
+
+namespace SeedLang.Interpreter {
+  // The SeedLang virtual machine to run bytecode stored in a chunk.
+  internal class VM {
+    private readonly VisualizerCenter _visualizerCenter;
+    private Chunk _chunk;
+    private VMValue[] _registers;
+
+    internal VM(VisualizerCenter visualizerCenter = null) {
+      _visualizerCenter = visualizerCenter ?? new VisualizerCenter();
+    }
+
+    internal void Run(Chunk chunk) {
+      _chunk = chunk;
+      _registers = new VMValue[chunk.RegisterCount];
+      int pc = 0;
```

Comment:
For the data type of PC, is `uint` better than `int`? I don't have a preference here. Just want to confirm that you have already considered all the corner cases. E.g. when a `JMP sBx` instruction results in a negative PC?